TAO/TAO_IDL/ast/ast_expression.cpp

```diff
@@ -3034,7 +2832,11 @@ dump_expr_val (ACE_OSTREAM_TYPE &o, AST_Expression::AST_ExprValue *ev)
       o << ev->u.wcval;
       break;
     case AST_Expression::EV_octet:
-      o << static_cast<int> (ev->u.oval);
+      {
+        char buffer[5] = {0};
+        ACE_OS::sprintf (buffer, "0x%02x", ev->u.oval);
+        o << &buffer[0];
```

Comment:
If this is going to an `ostream` would it be cleaner to use its own formatting capabilities instead of `sprintf`?

Reply:
I had to look everything up because I don't remember the last time I tried to use ostream formatting.

```c++
        std::ios saved(nullptr);
        saved.copyfmt(o);
        o << "0x" << std::hex << std::setw(2) << std::setfill('0') << unsigned(ev->u.oval);
        o.copyfmt(saved);
```

I'm not sure I'd call it cleaner. I'd definitely say much more verbose.

Reply:
Yes, it is more verbose... but it's using the object that's already involved in this operation instead of managing a separate buffer.